### DISCUSSION TITLE
Rmq client does not received the message in the order

### DISCUSSION THREAD

**Discussion by siugar** [rabbitmq/rabbitmq-server#11739] [Other] [answered]
**Rmq client does not received the message in the order.**

I am using .net RabbitMQ.Client 5.1.0 with the rmq server 3.7.8.
**I made a experiment to see the message order so I disconnected and reconnected my network during consuming the messages.**

There is only one exchnage, one queue and one channel for one consumer.

The function BasicAck(deliveryTag, true) was called periodially and PrefetchCount is 20.
I noticed the last BasicAck before I disconnected my network did not take the effect. (It's fine)

After I reconnected my network between Rmq server and client, the Rmq client recovered automatically and continued to received the messages.
Now the order of the messages is followed the last message I received actually before Rmq client recovered.
Later for seconds, the RESENT messages are received because I saw BasicDeliverEventArgs.Redelivered is true.
The 20 resent messages(Those were the unack messages during I disconnected the network) interrupted in the middle of the my received series.
The order is not what I think as the official website said. (https://www.rabbitmq.com/docs/semantics)

Is it normal or what I was misunderstanding?
Is it avoidable or the only way I have to do is to make the message processing Idempotence?

![order](https://github.com/user-attachments/assets/b1bd9ca8-07a7-48b0-975b-1017fd4fbae8)

**Comment by lukebakken** [comment]
Hello, RabbitMQ 3.7.8 is unsupported and VERY old. RabbitMQ.Client 5.1.0 is also unsupported.

If you can reproduce this issue using the current version of RabbitMQ (3.13.5), please do so and provide the code you are using to test.

**Comment by michaelklishin** [comment, marked as answer]
Those guarantees apply to initial deliveries. They do happen in order. 

Requeueing, redeliveries, QoS > 1 and concurrent dispatch of deliveries (to a thread pool or a task dispatcher or something like that) in client libraries all can affect what your consumer perceives as "delivery order".
